### DISCUSSION TITLE
class navigationHistory, path, snapshot

### DISCUSSION THREAD

**Discussion by torvista** [zencart/zencart#6837] [General]
While investigating in my site why a Back button does not link to expected, I've been watching (echo array in the header) a default site how the pages in the path array gets added to/subtracted from while navigating the information links, and I see previous (not the last) pages being removed, but I don't get the logic behind it.

Can someone explain what is supposed to be happening, and what is the purpose of the snapshot, which appears static with the last product viewed?

**Comment by drbyte** [comment]
The theory behind `navigationHistory` is to provide a stateful awareness of the "last" page viewed by the visitor so that we can generate our own "back" button as a navigation helper, usually at the bottom of a section's content, on pages where such navigation is sensible.
Additionally, the "snapshot" feature is used when we need to intercept the normal navigation in order to do something required before accessing the page requested, such as require a login. When the snapshot is needed, we use the snapshot as the "last" page, because we don't want a "back" button to take them back to something strange like "login". Basically it allows skipping of pages that wouldn't likely be what the "user" means by "back".

**Comment by torvista** [comment]
Thanks.

Responsive Classic, set Information box to top for ease of use.

Click on About Us: Back button points to index. OK.
Click on Contact Us: Back button points to About Us. OK.
Click on About Us: Back button points to index, instead of Contact Us. Is this expected behaviour?
